CDW autopopulator + readme file with instructions

ZubairL opened the pull request:
Description
The purpose of this ticket is to address comments in MCP-7625.

What was the problem?
N/A - to speed up populating the CDW Data Locations tab.

Jira Tickets
MCP-7582
MCP-7583


How to Test This PR
Choose any condition and clear the CDW Data Locations tab and run the script. Please see readme for further instructions.

jlpeterson2 reviewed (changes requested):
Hi Zubair! This is awesome, great work! A few changes: 
* It looks as if a medical code is assigned to multiple criteria, and any of those criteria does not occur by itself in the file, the script will not output that code and criteria. 
    * For example, in fibromyalgia, there is only one ICD-9 code for the diagnosis of fibro. That code happens to also be labeled with two other CFR criteria. After running the script, that row for ICD-9 does not appear 
    * Hemorrhoids has similar examples with "Internal Hemorrhoids," "External Hemorrhoids," etc. For this, I made a copy called "hem-test" 
* In cases where the CFR criteria includes another phrase from a different criteria, the code may pull in a wrong example. 
    *  This can also be seen in hemorrhoids (hem-test). There is criteria for "Hemorrhoids," "Internal Hemorrhoids," "External Hemorrhoids," and "Prolapsed Hemorrhoids." Since these criteria tend to have the same bug as before, the script is pulling in the wrong SNOMED-CT & ICD-9 codes for internal, external, and prolapsed. For example, the example code it is pulling for prolapsed is "70153002," which is just labeled as Hemorrhoids. 
    * This doesn't seem to be doing it for the ICD-10 codes
* In the readme, can you put a note that if there are DATETIME criteria, the user will need to manually enter those. 
    * These can be seen in some cancer and thrombo codes. Such as "100% rating continuation post-treatment"

Thanks!

ZubairL commented:
> Hi Zubair! This is awesome, great work! A few changes:
> 
> * It looks as if a medical code is assigned to multiple criteria, and any of those criteria does not occur by itself in the file, the script will not output that code and criteria.
>   
>   * For example, in fibromyalgia, there is only one ICD-9 code for the diagnosis of fibro. That code happens to also be labeled with two other CFR criteria. After running the script, that row for ICD-9 does not appear
>   * Hemorrhoids has similar examples with "Internal Hemorrhoids," "External Hemorrhoids," etc. For this, I made a copy called "hem-test"
> * In cases where the CFR criteria includes another phrase from a different criteria, the code may pull in a wrong example.
>   
>   * This can also be seen in hemorrhoids (hem-test). There is criteria for "Hemorrhoids," "Internal Hemorrhoids," "External Hemorrhoids," and "Prolapsed Hemorrhoids." Since these criteria tend to have the same bug as before, the script is pulling in the wrong SNOMED-CT & ICD-9 codes for internal, external, and prolapsed. For example, the example code it is pulling for prolapsed is "70153002," which is just labeled as Hemorrhoids.
>   * This doesn't seem to be doing it for the ICD-10 codes
> * In the readme, can you put a note that if there are DATETIME criteria, the user will need to manually enter those.
>   
>   * These can be seen in some cancer and thrombo codes. Such as "100% rating continuation post-treatment"
> 
> Thanks!




Interestingly enough, I think these errors are all happening for different reasons: 



>> Regarding the reason why that one Fibro code wasn't showing up: 
This had to do with the RegEx formula....think the script was getting confused because that criteria has parentheses. I've updated the ReGex to: 
 
let regex = new RegExp(`(^|; )${criteriaValue.replace(/[.*+?^${}()|[\]\\]/g, '\\$&')}($|; )`, "i");


Just a heads up when testing this for Fibro codes, the number of rows of the manually created CDW Data Locations and the automated one are different. In the manually created one, there are 4 ICD-9 codes that don't appear in the Code Set Details...not sure how those got there. 


>> Regarding the reason why some internal/external hemorrhoids didn't show up:
Similar to the reason above, but in our manually created CDW Data Locations I see that we have ICD-10 codes for both external and internal hemorrhoids, but these are actually not in the CDW. This explains why there are two less rows in the automated CDW Data locations sheet. 


>> Regarding the script pulling the wrong codes for Prolapsed, external, internal hemorrhoids:
This was a separate bug where Prolapsed, external, internal hemorrhoids was matching just hemorrhoids' codes. I've added a fix for this and it seems to be working fine, but please let me know if you catch it again with other conditions. 


>> Totally forgot about the DATETIME Criteria, I've gone ahead and added a note in the readme.

jlpeterson2 reviewed (approved):
This works great, thanks for the quick updates!

hannahbarker2 commented on utils/cdw_autopopulator.java:
```diff
@@ -0,0 +1,414 @@
+            columnFValue = "CPTCode, CPTName"
+
+        } else if (columnDValue === 'NDC') {
+            columnCValue = "NDC medication code"
```
Can you change this to "Medication NDC Code" to be consistent with the other labels?

hannahbarker2 reviewed (commented):
This is great, excellent work! Can you make the quick fix I suggested to ensure consistent labels? Will approve after that!

hannahbarker2 commented on utils/README_autopopulator.md:
```diff
@@ -0,0 +1,30 @@
+
+## Notes/Warnings/Tips
+- If there is any DATETIME criteria, the user will need to add this in manually. For examples of  DATETIME criteria, please take a look at some of our previous conditions such as Pleuritis, Thrombo or even GI Cancer. 
+- there are console log statements commented throughout the script. The user can use these to troubleshoot if there are any issues. 
```
Capitalize these bullet points under "Notes" for consistency

ZubairL replied:
@hannahbarker2 both these changes should be made now!

hannahbarker2 replied:
Thanks @ZubairL! Did you make changes to the UMLS SNOMED script? Wondering why those changes are included.
